ambry-replication/src/main/java/com/github/ambry/replication/ReplicaThread.java

```diff
@@ -348,6 +377,296 @@ protected Map<DataNodeId, List<RemoteReplicaInfo>> selectReplicas(Map<DataNodeId
     return replicas;
   }
 
+  /**
+   * Generates group ids for active iteration for replicas, all replicas are mapped to one group id only
+   * Generates group ids for all datanodes for standby no progress groups
+   * All group ids are unique
+   * All replicas are mapped to particular group id and each datanode has one stand by remote replica group id
+   * so we make less number of parallel calls to remote hosts
+   * @param groupIdToRemoteReplicaMap {@link Map} this map will be populated with group ids to replicas map
+   * @param standByNoProgressGroupId {@link Map}  this map will be populated with datanode to standby group id
+   */
+  void generateGroupIdsForReplicas(Map<Integer, List<RemoteReplicaInfo>> groupIdToRemoteReplicaMap,
```

Comment:
Use shorter names like create or make instead of generate.
If you're using get, then the fn must return something,

Reply:
changed to name.